hedera-node/src/main/java/com/hedera/services/txns/consensus/TopicUpdateTransitionLogic.java

```diff
@@ -21,281 +21,114 @@
-		try {
-			if (op.hasAdminKey()  && !applyNewAdminKey(op, topicId, topic)) {
-				return false;
-			}
-
-			if (op.hasSubmitKey()  && !applyNewSubmitKey(op)) {
-				return false;
-			}
-		} catch (DecoderException e) {
-			log.error("Decoder exception updating topic {}. ", topicId, e);
-			transactionContext.setStatus(BAD_ENCODING);
-			return false;
-		}
-		return true;
-	}
-
-	private boolean applyNewSubmitKey(final ConsensusUpdateTopicTransactionBody op) throws DecoderException {
-		var newSubmitKey = op.getSubmitKey();
-		if (!validator.hasGoodEncoding(newSubmitKey)) {
-			transactionContext.setStatus(BAD_ENCODING);
-			return false;
-		}
-		JKey.mapKey(newSubmitKey);
-		return true;
-	}
-
-	private boolean applyNewAdminKey(final ConsensusUpdateTopicTransactionBody op,
-			final TopicID topicId,
-			final MerkleTopic topic) throws DecoderException {
-		var newAdminKey = op.getAdminKey();
-		if (!validator.hasGoodEncoding(newAdminKey)) {
-			log.error("Update topic {} has invalid admin key specified, " +
-					"which should have been caught during signature validation", topicId);
-			transactionContext.setStatus(BAD_ENCODING);
-			return false;
-		}
-		var fcKey = JKey.mapKey(newAdminKey);
-		if (fcKey.isEmpty()) {
-			boolean opRemovesAutoRenewId = op.hasAutoRenewAccount() &&
-					designatesAccountRemoval(op.getAutoRenewAccount());
-			if (topic.hasAutoRenewAccountId() && !opRemovesAutoRenewId) {
-				transactionContext.setStatus(AUTORENEW_ACCOUNT_NOT_ALLOWED);
-				return false;
-			}
-		}
-		return true;
-	}
-
-	private void applyNewKeys(ConsensusUpdateTopicTransactionBody op, MerkleTopic topic) {
-		if (op.hasAdminKey()) {
-			topic.setAdminKey(asFcKeyUnchecked(op.getAdminKey()));
-		}
-		if (op.hasSubmitKey()) {
-			topic.setSubmitKey(asFcKeyUnchecked(op.getSubmitKey()));
-		}
+		/* --- Extract gRPC --- */
+		final var transactionBody = transactionContext.accessor().getTxn();
+		final var op = transactionBody.getConsensusUpdateTopic();
+		final var topicId = op.getTopicID();
+		final var newAdminKey = op.hasAdminKey() ? op.getAdminKey() : null;
+		final var newSubmitKey = op.hasSubmitKey() ? op.getSubmitKey() : null;
+		final var newMemo = op.hasMemo() ? op.getMemo().getValue() : null;
+		final var newExpirationTime = op.hasExpirationTime() ? op.getExpirationTime() : null;
+		final var newAutoRenewPeriod = op.hasAutoRenewPeriod() ? op.getAutoRenewPeriod() : null;
+		final var topic = topicStore.loadTopic(Id.fromGrpcTopic(topicId));
+
+		final var affectsExpiryOnly = !op.hasMemo()
+				&& !op.hasAdminKey()
+				&& !op.hasSubmitKey()
+				&& !op.hasAutoRenewPeriod()
+				&& !op.hasAutoRenewAccount();
+
+		JKey newAdminKeyDecoded = null;
+
+		/* --- Validate --- */
+		validateFalse(!topic.hasAdminKey() && !affectsExpiryOnly, UNAUTHORIZED);
+		if (newAdminKey != null) {
+			validateFalse(!validator.hasGoodEncoding(newAdminKey), BAD_ENCODING);
+			newAdminKeyDecoded = validator.attemptDecodeOrThrow(newAdminKey);
+		}
```

Comment:
- It seems that we are performing validation on the `newAdminKey` in the `validatePreSignatureValidation` method. Why would we need to do it again? 
- I think that we do not need to do `validator.attemptDecodeOrThrow(newAdminKey);` and create instance of `newAdminKeyDecoded` here. We can have the same implementation as `newSubmitKey` where we pass it directly to the `.update` and do `asFcKeyUnchecked(k)`